docs/decisions/0045-updated-vector-store-design.md

```diff
@@ -0,0 +1,811 @@
+}
+
+class AzureAISearchMemoryCollectionService: IMemoryCollectionService;
+class RedisMemoryCollectionService: IMemoryCollectionService;
+class WeaviateMemoryCollectionService: IMemoryCollectionService;
+
+// Customers can inherit from our implementations and replace just the creation scenarios to match their schemas.
+class CustomerCollectionsService: AzureAISearchMemoryCollectionService, IMemoryCollectionService;
+
+// We can also create implementations that create indices based on an MLIndex specification.
+class MLIndexAzureAISearchMemoryCollectionService(MLIndex mlIndexSpec): AzureAISearchMemoryCollectionService, IMemoryCollectionService;
+
+interface IMemoryRecordService<TDataModel>
+{
+    Task<TDataModel?> GetAsync(string key, GetRecordOptions? options = default, CancellationToken cancellationToken = default);
+    Task DeleteAsync(string key, DeleteRecordOptions? options = default, CancellationToken cancellationToken = default);
+    Task<string> UpsertAsync(TDataModel record, UpsertRecordOptions? options = default, CancellationToken cancellationToken = default);
+}
+
+class AzureAISearchMemoryRecordService<TDataModel>(): IMemoryRecordService<TDataModel>;
+```
+
+#### Option 3 - Separated collection and record management with collection create separate from other operations.
+
+Vector store same as option 2 so not repeated for brevity.
+
+```cs
+
+interface IMemoryCollectionCreateService
+{
+    virtual Task CreateCollectionAsync(string name, CancellationToken cancellationToken = default);
+}
+
+class AzureAISearchChatHistoryCollectionCreateService: IMemoryCollectionCreateService;
+class AzureAISearchSemanticCacheCollectionCreateService: IMemoryCollectionCreateService;
+
+// Customers can create their own creation scenarios to match their schemas, but can continue to use our get, does exist and delete class.
+class CustomerChatHistoryCollectionCreateService: IMemoryCollectionCreateService;
+
+interface IMemoryCollectionUpdateService
+{
+    IAsyncEnumerable<string> ListCollectionNamesAsync(CancellationToken cancellationToken = default);
+    Task<bool> CollectionExistsAsync(string name, CancellationToken cancellationToken = default);
+    Task DeleteCollectionAsync(string name, CancellationToken cancellationToken = default);
+}
+
+class AzureAISearchMemoryCollectionUpdateService: IMemoryCollectionUpdateService;
+class RedisMemoryCollectionUpdateService: IMemoryCollectionUpdateService;
+class WeaviateMemoryCollectionUpdateService: IMemoryCollectionUpdateService;
+
+```
+
+#### Option 4 - Separated collection and record management with collection create separate from other operations, with collection management aggregation class on top.
+
+Variation on option 3. 
+
+```cs
+
+interface IMemoryCollectionCreateService
+{
+    virtual Task CreateCollectionAsync(string name, CancellationToken cancellationToken = default);
+}
+
+interface IMemoryCollectionUpdateService
```

Comment:
Thanks for the feedback.  My reasoning was in the Decision Outcome section, but I added more detail just now, to try and explain it in more detail.

The main reason for splitting out collection create is because I expect that there will be multiple implementations of this, even for the same database type, while list, exists and delete are the same across all and will only require a single implementation.